src/app/routes/api/v3/public/__tests__/public-forms.routes.spec.ts

```diff
@@ -0,0 +1,1140 @@
+import { MyInfoGovClient } from '@opengovsg/myinfo-gov-client'
+import SPCPAuthClient from '@opengovsg/spcp-auth-client'
+import { getReasonPhrase, StatusCodes } from 'http-status-codes'
+import { omit } from 'lodash'
+import mongoose from 'mongoose'
+import { errAsync } from 'neverthrow'
+import supertest, { Session } from 'supertest-session'
+import { mocked } from 'ts-jest/utils'
+
+import { DatabaseError } from 'src/app/modules/core/core.errors'
+import {
+  MOCK_COOKIE_AGE,
+  MOCK_UINFIN,
+} from 'src/app/modules/myinfo/__tests__/myinfo.test.constants'
+import { MYINFO_COOKIE_NAME } from 'src/app/modules/myinfo/myinfo.constants'
+import { MyInfoCookieState } from 'src/app/modules/myinfo/myinfo.types'
+import getMyInfoHashModel from 'src/app/modules/myinfo/myinfo_hash.model'
+import { AuthType, IFieldSchema, Status } from 'src/types'
+
+import { setupApp } from 'tests/integration/helpers/express-setup'
+import { buildCelebrateError } from 'tests/unit/backend/helpers/celebrate'
+import dbHandler from 'tests/unit/backend/helpers/jest-db'
+
+import * as FormService from '../../../../../modules/form/form.service'
+
+import {
+  MOCK_ATTACHMENT_FIELD,
+  MOCK_ATTACHMENT_RESPONSE,
+  MOCK_CHECKBOX_FIELD,
+  MOCK_CHECKBOX_RESPONSE,
+  MOCK_NO_RESPONSES_BODY,
+  MOCK_OPTIONAL_VERIFIED_FIELD,
+  MOCK_OPTIONAL_VERIFIED_RESPONSE,
+  MOCK_SECTION_FIELD,
+  MOCK_SECTION_RESPONSE,
+  MOCK_TEXT_FIELD,
+  MOCK_TEXTFIELD_RESPONSE,
+} from './public-forms.routes.spec.constants'
+
+const MyInfoHashModel = getMyInfoHashModel(mongoose)
+
+jest.mock('@opengovsg/spcp-auth-client')
+const MockAuthClient = mocked(SPCPAuthClient, true)
+
+jest.mock('nodemailer', () => ({
+  createTransport: jest.fn().mockReturnValue({
+    sendMail: jest.fn().mockResolvedValue(true),
+  }),
+}))
+
+jest.mock('@opengovsg/myinfo-gov-client', () => ({
+  MyInfoGovClient: jest.fn(),
+  MyInfoMode: jest.requireActual('@opengovsg/myinfo-gov-client').MyInfoMode,
+  MyInfoSource: jest.requireActual('@opengovsg/myinfo-gov-client').MyInfoSource,
+  MyInfoAddressType: jest.requireActual('@opengovsg/myinfo-gov-client')
+    .MyInfoAddressType,
+  MyInfoAttribute: jest.requireActual('@opengovsg/myinfo-gov-client')
+    .MyInfoAttribute,
+}))
+const MockMyInfoGovClient = mocked(MyInfoGovClient, true)
+const mockExtractUinFin = jest.fn()
+MockMyInfoGovClient.mockImplementation(
+  () =>
+    (({
+      extractUinFin: mockExtractUinFin,
+    } as unknown) as MyInfoGovClient),
+)
+
+// Import last so mocks are imported correctly
+// eslint-disable-next-line import/first
+import { PublicFormsRouter } from '../public-forms.routes'
```

Comment:
this probably does not matter anymore, can move the import to its correct position and remove the comment and lint ignore line

Reply:
this does matter and removing the import causes the successful `MyInfo` test to fail. This is because the `mockExtractUinFin`, which needs to be setup in order for the test to pass successfully.